fastly/block_fastly_service_logging_newrelicotlp.go

```diff
@@ -0,0 +1,264 @@
+package fastly
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	gofastly "github.com/fastly/go-fastly/v8/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// NewRelicOTLPServiceAttributeHandler provides a base implementation for ServiceAttributeDefinition.
+type NewRelicOTLPServiceAttributeHandler struct {
+	*DefaultServiceAttributeHandler
+}
+
+// NewServiceLoggingNewRelicOTLP returns a new resource.
+func NewServiceLoggingNewRelicOTLP(sa ServiceMetadata) ServiceAttributeDefinition {
+	return ToServiceAttributeDefinition(&NewRelicOTLPServiceAttributeHandler{
+		&DefaultServiceAttributeHandler{
+			key:             "logging_newrelicotlp",
+			serviceMetadata: sa,
+		},
+	})
+}
+
+// Key returns the resource key.
+func (h *NewRelicOTLPServiceAttributeHandler) Key() string {
+	return h.key
+}
+
+// GetSchema returns the resource schema.
+func (h *NewRelicOTLPServiceAttributeHandler) GetSchema() *schema.Schema {
+	blockAttributes := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The unique name of the New Relic OTLP logging endpoint. It is important to note that changing this attribute will delete and recreate the resource",
+		},
+		"region": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "US",
+			Description: "The region that log data will be sent to. Default: `US`",
+		},
+		"token": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Sensitive:   true,
+			Description: "The Insert API key from the Account page of your New Relic account",
+		},
+	}
+
+	if h.GetServiceMetadata().serviceType == ServiceTypeVCL {
```

Comment:
As this is only used for a VCL service, let's remove this `if` statement and inline each of its block attributes into `blockAttributes` above.